stats/statsview.cpp

```diff
@@ -1021,12 +1028,21 @@ static bool is_linear_regression(int sample_size, double cov, double sx2, double
 	return true; // can't happen, as we tested for sample_size above.
 }
 
-// Returns the coefficients [a,b] of the line y = ax + b
-// If case of an undetermined regression or one with infinite slope, returns [nan, nan]
-static std::pair<double, double> linear_regression(const std::vector<StatsScatterItem> &v)
+struct regression_data {
+	double a,b;
+	double res2, r2;
+};
+
+// Returns the coefficients a,b of the line y = ax + b
+// as well as the variance of the residuals (averaged residual squared) as res2
+// and r^2 = 1.0 - variance of data / res2 which is the fraction of the variance of
+// the data that is explained by the linear regression.
+// If case of an undetermined regression or one with infinite slope, returns {nan, nan, 0.0, 0.0}
+
+static struct regression_data linear_regression(const std::vector<StatsScatterItem> &v)
 {
 	if (v.size() < 2)
-		return { NaN, NaN };
+		return { .a = NaN, .b = NaN, .res2 = 0.0, .r2 = 0.0};
```

Comment:
Note that this syntax is C++20 only. Somewhat surprisingly, it appears to work with our compilers.